staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal"
+	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
+)
+
+var (
+	scaleGroupVersion   = schema.GroupVersion{Group: "autoscaling", Version: "v1"}
+	replicasPathInScale = fieldpath.MakePathOrDie("spec", "replicas")
+)
+
+// ResourcePathMappings maps a group/version to its replicas path
+type ResourcePathMappings map[string]fieldpath.Path
+
+// ScaleHandler manages the conversion of managed fields between a main
```

Comment:
Since this code is in a very generic place, IMO it should be named for what it does, not what you intend to use it for. E.g. this is a "simple mapped-path handler" or something like that. (not sure if I'm asking for a change yet or not)

Reply:
True, the `struct` is potentially very generic. On the other hand we only manage scale throughout its logic. I think making more specific has value here because otherwise the number of edge cases explode.

Reply:
I also prefer keeping it specific--but then I'm not sure it should be located here?

Reply:
We'll want to use it for CRDs and other types of resources. And we wanted to keep the "manager" manipulation within that package.